docs/src/modules/sandbox/Dependencies.ts

```diff
@@ -0,0 +1,172 @@
+import { CODE_VARIANTS } from '../constants';
+
+export default function SandboxDependencies(
+  demo: {
+    raw: string;
+    product?: 'joy-ui' | 'base';
+    codeVariant: keyof typeof CODE_VARIANTS;
+  },
+  options?: { commitRef?: string },
+) {
+  const { commitRef } = options || {};
+
+  /**
+   * WARNING: Always uses `latest` typings.
+   *
+   * Adds dependencies to @types packages only for packages that are not listed
+   * in packagesWithBundledTypes
+   *
+   * @param deps - list of dependency as `name => version`
+   */
+  function addTypeDeps(deps: Record<string, string>): void {
+    const packagesWithBundledTypes = ['date-fns', '@emotion/react', '@emotion/styled'];
+    const packagesWithDTPackage = Object.keys(deps)
+      .filter((name) => packagesWithBundledTypes.indexOf(name) === -1)
+      // All the MUI packages come with bundled types
+      .filter((name) => name.indexOf('@mui/') !== 0);
+
+    packagesWithDTPackage.forEach((name) => {
+      let resolvedName = name;
+      // scoped package?
+      if (name.startsWith('@')) {
+        // https://github.com/DefinitelyTyped/DefinitelyTyped#what-about-scoped-packages
+        resolvedName = name.slice(1).replace('/', '__');
+      }
+
+      deps[`@types/${resolvedName}`] = 'latest';
+    });
+  }
+
+  /**
+   * @param packageName - The name of a package living inside this repository.
+   * @return string - A valid version for a dependency entry in a package.json
+   */
+  function getMuiPackageVersion(packageName: string): string {
+    if (
+      commitRef === undefined ||
+      process.env.SOURCE_CODE_REPO !== 'https://github.com/mui/material-ui'
+    ) {
+      // #default-branch-switch
+      return 'latest';
+    }
+    const shortSha = commitRef.slice(0, 8);
+    return `https://pkg.csb.dev/mui/material-ui/commit/${shortSha}/@mui/${packageName}`;
+  }
+
+  function extractDependencies(raw: string) {
+    function includePeerDependencies(
+      deps: Record<string, string>,
+      versions: Record<string, string>,
+    ): Record<string, string> {
+      let newDeps: Record<string, string> = {
+        ...deps,
+        'react-dom': versions['react-dom'],
+        react: versions.react,
+        '@emotion/react': versions['@emotion/react'],
+        '@emotion/styled': versions['@emotion/styled'],
+      };
+
+      if (newDeps['@mui/lab']) {
+        newDeps['@mui/material'] = versions['@mui/material'];
+      }
+
+      if (newDeps['@material-ui/data-grid']) {
```

Comment:
We can update this for https://mui.com/material-ui/react-table/#data-table

```suggestion
      if (newDeps['@mui/x-data-grid']) {
```

Reply:
Updated.